### PR TITLE
[dashboard] don't show 'new project' button

### LINKED ISSUE #5915: [dashboard] workspaces wrongly shows "new project" button
On a personal account when there are no projects, I see the "new project" button on the /workspaces list.

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
There is never a case where we want to show "New Project" on the top right of the list, because in this situation there are already workspaces and hence we know who to start more of them.

## Related Issue(s)

Fixes #5915

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **svenefftinge** after the PR has been reviewed.

Associated issue: *#5915*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Review by gtsiolis — APPROVED** [review]
Whoa! Nice catch! ⚾

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 0e5e56385b7954758ba25116e981b5c8d974e348</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5916#pullrequestreview-766407406" title="Approved">gtsiolis</a>*

Associated issue: *#5915*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
